config/channel/post-install/400-kafkachannel-update-job.yaml

```diff
@@ -0,0 +1,49 @@
+# Copyright 2021 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: eventing-kafka-v0.26-channel-post-install-job
```

Comment:
should we use `generatedName`?  (which would require `k create`, instead of `k apply`)

Reply:
> should we use generatedName? (which would require k create, instead of k apply)

Hmmm... I'm not sure what the best practice is in that regard?   From a quick internet scan I see the following...

**PROs...**
- Can re-run Job without first deleting old instance

**CONs...**
- Have to use `create` instead of `apply`
- Does not work with Helm (See [Issue](https://github.com/helm/helm/issues/3348#issuecomment-482369133))

Knative itself doesn't seem to have settled on anything, I saw the use of `generateName` in a Serving post-install job, but not in the most-recent Eventing post-install job (nor in most-recent `eventing-kafka` [post-install job](https://github.com/knative-sandbox/eventing-kafka/pull/615)).

I vote to leave it as is given the downsides and lack of a Knative standard.  As an example of potential user impact - we (SAP) would have to make a small change in some of our tooling to use `create` instead of `apply`.  If you feel strongly though I'm happy to change it.